distribution/docker/src/test/java/org/elasticsearch/docker/test/DockerYmlTestSuiteIT.java

```diff
@@ -0,0 +1,163 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.docker.test;
+
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.common.CharArrays;
+import org.elasticsearch.common.io.PathUtils;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.test.rest.ESRestTestCase;
+import org.elasticsearch.test.rest.yaml.ClientYamlTestCandidate;
+import org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.CharBuffer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Base64;
+
+public class DockerYmlTestSuiteIT extends ESClientYamlSuiteTestCase {
+
+    private static final String USER = "x_pack_rest_user";
+    private static final String PASS = "x-pack-test-password";
+    private static final String KEYSTORE_PASS = "testnode";
+
+    public DockerYmlTestSuiteIT(ClientYamlTestCandidate testCandidate) {
+        super(testCandidate);
+    }
+
+    @ParametersFactory
+    public static Iterable<Object[]> parameters() throws Exception {
+        return createParameters();
+    }
+
+    @Override
+    protected String getTestRestCluster() {
+        String distribution = getDistribution();
+        return new StringBuilder()
+            .append("localhost:")
+            .append(getProperty("test.fixtures.elasticsearch-" + distribution + "-1.tcp.9200"))
+            .append(",")
+            .append("localhost:")
+            .append(getProperty("test.fixtures.elasticsearch-" + distribution + "-2.tcp.9200"))
+            .toString();
+    }
+
+    @Override
+    protected boolean randomizeContentType() {
+        return false;
+    }
+
+    private String getDistribution() {
+        String distribution = System.getProperty("tests.distribution", "default");
+        if (distribution.equals("oss") == false && distribution.equals("default") == false) {
+            throw new IllegalArgumentException("supported values for tests.distribution are oss or default but it was " + distribution);
+        }
+        return distribution;
+    }
+
+    private boolean isOss() {
+        return getDistribution().equals("oss");
+    }
+
+    private String getProperty(String key) {
```

Comment:
This method seems misnamed if the expectation is the key produces a port.

Reply:
the variable name was confusing, the expectation is that it's present - and it happens to be a port, but that's not the concern of this method. I renamed the variable.